### ISSUE TITLE
Compatibility: Remove dart:io dependency from generated Client

### ISSUE THREAD

**Issue by tgrushka** [serverpod/serverpod#1338]
**Is your feature request related to a problem? Please describe.**

I am trying out jaspr (https://pub.dev/packages/jaspr) to build a "Flutter-like" HTML/CSS web site frontend for my serverpod app (because it compiles to 1-2 MB instead of 20 MB). In order to import my project_client into this dart-js app and build to JavaScript, it cannot depend on dart:io.

I noticed my generated `project_client/lib/src/protocol/client.dart` depends on dart:io:
```dart
import 'dart:io' as _i4;
...
  Client(
    String host, {
    _i4.SecurityContext? context, // depends on dart:io/SecurityContext
    _i1.AuthenticationKeyManager? authenticationKeyManager,
  }) : super(
...
```

The only workaround is to edit the above code in the generated file to make `context` a `dynamic`, and remove the `_i4` / `dart:io` import line.

However, the `ServerpodClient` abstract super class allows the `context` to be `dynamic`:
```dart
  ServerpodClient(
    String host,
    SerializationManager serializationManager, {
    dynamic context,
...
```

**Describe the solution you'd like**
In `serverpod_cli/lib/src/generator/dart/library_generators/library_generator.dart` we are explicitly requiring this dart:io dependency in the declared parameter:
```dart
                    Parameter((p) => p
                      ..name = 'context'
                      ..named = true
                      ..type = TypeReference((t) => t
                        ..symbol = 'SecurityContext'
                        ..url = 'dart:io'
                        ..isNullable = true)),
```

I propose changing this to `dynamic`, as it is `dynamic` in the super class, and in `serverpod_client/lib/src/serverpod_client_io.dart`, we are checking the type of `context` dynamically, anyway. Proposed change:

```dart
                    Parameter((p) => p
                      ..name = 'context'
                      ..named = true
                      ..type = TypeReference((t) => t
                        ..symbol = 'dynamic')),
```

@vlidholt I see that you implemented this, do you know if making `context` a `dynamic` in the generated `Client` would be acceptable, or cause any issues in the future roadmap?

**Describe alternatives you've considered**
None that I know of, as we are not generating a separate dart:io and dart:html `Client`. That could be an alternative solution but perhaps too complicated, unless it's on the roadmap in the future. If we were to go that route, hopefully the Client could still be imported in one line and the platform checking could be done in the generated library.

**Additional context**
serverpod v1.1.1 / main branch

**Comment by Maksimka101** [comment]
It's fixed in the serverpod 1.2.6 or even earlier. @vlidholt you can close this issue! :)

**Comment by SandPod** [comment]
Closing this issue, thanks for the ping @Maksimka101 🙏
